### ISSUE TITLE
Half-Life 2 (Update): Crashes upon loading in DX9 mode and shadow options/color correction missing

### ISSUE THREAD

**Issue by RavenMacDaddy** [dege-diosg/dgVoodoo2#91]
Hello

So I was trying to wrap HL2 to use Auto HDR, but if I run the game with DX9 (the default which is v9.5 if I understand things correctly), it will throw up an error upon loading a game:

![image](https://user-images.githubusercontent.com/67732686/137310089-ff95d6c0-edc3-46f3-b6d5-324e593b7a34.png)

I also noticed I couldn't choose shadow details above Medium.

Reading up on that specifically, it seems like the game wraps the game to DX11 with the d3d9.dll, then gets confused and somehow expects DX8 to be used - hence the error above, and because DX8 mode for the game is limited to shadow details going up to Medium, as well as not having Color Correction (or motion blur).

Even more weird is that, if I actually force the game to use only DX8 by applying the launch option -dxlevel 81 or 80 (81 being the highest DX8 API available), it will still somehow make use of DX9 (DX9 is always the API presented by the RTSS OSD if dgVoodoo isn't being used), so if I don't provide the d3d9.dll, it won't wrap up to DX11 and d3d8.dll is never needed.

By the way, the game loads any DLLs from its _bin_ dir, and providing additional DLLs like d3d8 from dgVoodoo doesn't make a difference since it's seemingly never used for anything.

---

To summarize:

- DLLs are loaded from the game's bin dir.
- The game only needs d3d9.dll whether running in DX8 or 9 mode and providing additional DLLs from dgVoodoo doesn't make a difference, because, for whatever reason, the game always uses DX9 for presentation it seems like - only the feature set changes.
- If you run the game with the default DX9 mode, it will crash with the error in the image above upon loading a game.
- The game is always stuck with the DX8 feature set while using dgVoodoo, namely shadow details go up to Medium and Color Correction is unavailable.
